### ISSUE TITLE
podman run returns error "unexpected end of JSON input"

### ISSUE THREAD

**Issue by VeitSanner** [containers/podman#11837]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

When I try to run a container using `podman v3.4.0` on macOS 11.6 x86_64, I get the following issue:

```bash
$ podman run hello-world 
Resolved "hello-world" as an alias (/etc/containers/registries.conf.d/000-shortnames.conf)
Trying to pull docker.io/library/hello-world:latest...
Getting image source signatures
Copying blob sha256:2db29710123e3e53a794f2694094b9b4338aa9ee5c40b930cb8063a1be392c54
Copying blob sha256:2db29710123e3e53a794f2694094b9b4338aa9ee5c40b930cb8063a1be392c54
Copying config sha256:feb5d9fea6a5e9606aa995e879d862b825965ba48de054caab5ef356dc6b3412
Writing manifest to image destination
Storing signatures
Error: error preparing container 4c55ffed4a30a40b87a20cffc862661c8755c81d766b6cac45e0ee680d64f2fe for attach: error configuring network namespace for container 4c55ffed4a30a40b87a20cffc862661c8755c81d766b6cac45e0ee680d64f2fe: error adding pod busy_darwin_busy_darwin to CNI network "podman": unexpected end of JSON input
```

Running a container fails for `podman remote` as well as when I ssh into the machine and execute the command with the same error.

**Describe the results you received:**

Running a container fails with the error message:

```text
Error: error preparing container 4c55ffed4a30a40b87a20cffc862661c8755c81d766b6cac45e0ee680d64f2fe for attach: error configuring network namespace for container 4c55ffed4a30a40b87a20cffc862661c8755c81d766b6cac45e0ee680d64f2fe: error adding pod busy_darwin_busy_darwin to CNI network "podman": unexpected end of JSON input
```

**Describe the results you expected:**
The container starts.

**Additional information you deem important (e.g. issue happens only occasionally):**

Looking at the symptom this looks similar to #11452 and  #11235, but I already checked and I do not have a `~/.docker` directory anymore.

```text
$ podman network inspect podman
```
 Result
```json
[
    {
        "cniVersion": "0.4.0",
        "name": "podman",
        "plugins": [
            {
                "bridge": "cni-podman0",
                "hairpinMode": true,
                "ipMasq": true,
                "ipam": {
                    "ranges": [
                        [
                            {
                                "gateway": "10.88.0.1",
                                "subnet": "10.88.0.0/16"
                            }
                        ]
                    ],
                    "routes": [
                        {
                            "dst": "0.0.0.0/0"
                        }
                    ],
                    "type": "host-local"
                },
                "isGateway": true,
                "type": "bridge"
            },
            {
                "capabilities": {
                    "portMappings": true
                },
                "type": "podman-machine"
            },
            {
                "capabilities": {
                    "portMappings": true
                },
                "type": "portmap"
            },
            {
                "type": "firewall"
            },
            {
                "type": "tuning"
            }
        ]
    }
]
```

**Output of `podman version`:**

```
Client:
Version:      3.4.0
API Version:  3.4.0
Go Version:   go1.17.1
Built:        Thu Sep 30 20:44:31 2021
OS/Arch:      darwin/amd64

Server:
Version:      3.3.1
API Version:  3.3.1
Go Version:   go1.16.6
Built:        Mon Aug 30 22:46:36 2021
OS/Arch:      linux/amd64
```

**Output of `podman info --debug`:**

```
host:
  arch: amd64
  buildahVersion: 1.22.3
  cgroupControllers: []
  cgroupManager: systemd
  cgroupVersion: v2
  conmon:
    package: conmon-2.0.29-2.fc34.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.0.29, commit: '
  cpus: 1
  distribution:
    distribution: fedora
    version: "34"
  eventLogger: journald
  hostname: localhost
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
  kernel: 5.13.16-200.fc34.x86_64
  linkmode: dynamic
  logDriver: ""
  memFree: 3644575744
  memTotal: 4104728576
  ociRuntime:
    name: crun
    package: crun-1.0-1.fc34.x86_64
    path: /usr/bin/crun
    version: |-
      crun version 1.0
      commit: 139dc6971e2f1d931af520188763e984d6cdfbf8
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +YAJL
  os: linux
  remoteSocket:
    exists: true
    path: /run/user/1000/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: true
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: true
  serviceIsRemote: true
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns-1.1.12-2.fc34.x86_64
    version: |-
      slirp4netns version 1.1.12
      commit: 7a104a101aa3278a2152351a082a6df71f57c9a3
      libslirp: 4.4.0
      SLIRP_CONFIG_VERSION_MAX: 3
      libseccomp: 2.5.0
  swapFree: 0
  swapTotal: 0
  uptime: 20m 17.49s
plugins:
  log: null
  network: null
  volume: null
registries:
  search:
  - docker.io
store:
  configFile: /var/home/core/.config/containers/storage.conf
  containerStore:
    number: 11
    paused: 0
    running: 0
    stopped: 11
  graphDriverName: overlay
  graphOptions: {}
  graphRoot: /var/home/core/.local/share/containers/storage
  graphStatus:
    Backing Filesystem: xfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
    Using metacopy: "false"
  imageStore:
    number: 4
  runRoot: /run/user/1000/containers
  volumePath: /var/home/core/.local/share/containers/storage/volumes
version:
  APIVersion: 3.3.1
  Built: 1630356396
  BuiltTime: Mon Aug 30 20:46:36 2021
  GitCommit: ""
  GoVersion: go1.16.6
  OsArch: linux/amd64
  Version: 3.3.1
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
$ brew list podman
/usr/local/Cellar/podman/3.4.0/bin/podman
/usr/local/Cellar/podman/3.4.0/bin/podman-remote
/usr/local/Cellar/podman/3.4.0/etc/bash_completion.d/podman
/usr/local/Cellar/podman/3.4.0/libexec/gvproxy
/usr/local/Cellar/podman/3.4.0/share/fish/vendor_completions.d/podman.fish
/usr/local/Cellar/podman/3.4.0/share/man/ (160 files)
/usr/local/Cellar/podman/3.4.0/share/zsh/site-functions/_podman
```

**Have you tested with the latest version of Podman and have you checked the Podman Troubleshooting Guide? (https://github.com/containers/podman/blob/master/troubleshooting.md)**

Yes

**Additional environment details (AWS, VirtualBox, physical, etc.):**

**Comment by mgorsk1** [comment]
+1, experiencing the same issue on fresh install: 

```
➜  ~ podman info
host:
  arch: amd64
  buildahVersion: 1.22.3
  cgroupControllers: []
  cgroupManager: systemd
  cgroupVersion: v2
  conmon:
    package: conmon-2.0.29-2.fc34.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.0.29, commit: '
  cpus: 1
  distribution:
    distribution: fedora
    version: "34"
  eventLogger: journald
  hostname: localhost
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
  kernel: 5.13.16-200.fc34.x86_64
  linkmode: dynamic
  logDriver: ""
  memFree: 1335873536
  memTotal: 2061852672
  ociRuntime:
    name: crun
    package: crun-1.0-1.fc34.x86_64
    path: /usr/bin/crun
    version: |-
      crun version 1.0
      commit: 139dc6971e2f1d931af520188763e984d6cdfbf8
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +YAJL
  os: linux
  remoteSocket:
    exists: true
    path: /run/user/1000/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: true
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: true
  serviceIsRemote: true
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns-1.1.12-2.fc34.x86_64
    version: |-
      slirp4netns version 1.1.12
      commit: 7a104a101aa3278a2152351a082a6df71f57c9a3
      libslirp: 4.4.0
      SLIRP_CONFIG_VERSION_MAX: 3
      libseccomp: 2.5.0
  swapFree: 0
  swapTotal: 0
  uptime: 5m 32.04s
plugins:
  log: null
  network: null
  volume: null
registries:
  search:
  - docker.io
store:
  configFile: /var/home/core/.config/containers/storage.conf
  containerStore:
    number: 6
    paused: 0
    running: 0
    stopped: 6
  graphDriverName: overlay
  graphOptions: {}
  graphRoot: /var/home/core/.local/share/containers/storage
  graphStatus:
    Backing Filesystem: xfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
    Using metacopy: "false"
  imageStore:
    number: 1
  runRoot: /run/user/1000/containers
  volumePath: /var/home/core/.local/share/containers/storage/volumes
version:
  APIVersion: 3.3.1
  Built: 1630356396
  BuiltTime: Mon Aug 30 20:46:36 2021
  GitCommit: ""
  GoVersion: go1.16.6
  OsArch: linux/amd64
  Version: 3.3.1
➜  ~ podman run hello-world
Error: error preparing container 7413f0d7def149c20c2beb5d35b0593db342b7a1b97fd5c3e1e56646f0f52319 for attach: error configuring network namespace for container 7413f0d7def149c20c2beb5d35b0593db342b7a1b97fd5c3e1e56646f0f52319: error adding pod relaxed_curie_relaxed_curie to CNI network "podman": unexpected end of JSON input
```

**Comment by afbjorklund** [comment]
It's a bug in podman-plug-ins (on the server), that bundles podman-machine-cni 0.2

**Comment by afbjorklund** [comment]
I think 3.3.1 helps, because it runs experimental versions of CoreOS ? ("next")

**Comment by Luap99** [comment]
Duplicate of #11413 
Should be fixed once podman 3.4 lands in CoreOS, as workaround you have to forward at least one port, e.g. `-p 8080`.

**Comment by VeitSanner** [comment]
@Luap99 I tried to run with `podman run -P hello-world ` but no joy.

```shell
Error: error preparing container 89fed3a2d595dc28c4c5ce1af577b438a19457bcf0ddd9e1b19f9e658074d7aa for attach: error configuring network namespace for container 89fed3a2d595dc28c4c5ce1af577b438a19457bcf0ddd9e1b19f9e658074d7aa: error adding pod jovial_raman_jovial_raman to CNI network "podman": unexpected end of JSON input
```

**Comment by Luap99** [comment]
`-P` only publishes ports which are defined in the image, the hello-world one has none.
use `-p 8080` or some other random port number

**Comment by afbjorklund** [comment]
I suppose there is no way of getting a fixed podman-plugins rpm out ? 3.3.1-2

**Comment by bretonics** [comment]
@Luap99 any idea when podman 3.4 will land in CoreOS? Current experiencing this issue on a fresh podman install

**Comment by afbjorklund** [comment]
Fedora CoreOS updates every two weeks (14 days): https://getfedora.org/en/coreos?stream=testing - see #11680

**Comment by willcohen** [comment]
It looks like that 3.4 is still working its way into both Fedora 34 and 35, which I assume needs to happen before FCOS picks it up:

https://bodhi.fedoraproject.org/updates/?search=podman

**Comment by amyruski** [comment]
Any update on when this fix will be available for download from homebrew?

**Comment by afbjorklund** [comment]
I don't think `brew` is involved, podman machine will auto-update CoreOS

**Comment by afbjorklund** [comment]
It will update every two weeks, you can check in the os-release (with ssh)

podman 3.3.1
Fedora CoreOS 35.**20211003**.1.0
https://getfedora.org/en/coreos?stream=next

podman 3.4.0
Fedora CoreOS 34.**20211004**.2.0
https://getfedora.org/en/coreos?stream=testing

This is supposed to be documented somewhere, and link to Fedora CoreOS

* Issue #11680

**Comment by afbjorklund** [comment]
Where "newer" is supposed to be "better" ? (In this case, "older" would have been preferred).

I was playing with using Fedora Cloud for podman instead of Fedora CoreOS, while it is broken.

**Comment by bavardage** [comment]
just a +1 that I encountered the same issue "error configuring network namespace for container ..." and the workaround fixed it

**Comment by jgowdy** [comment]
If you're on M1 https://builds.coreos.fedoraproject.org/prod/streams/testing-devel/builds/34.20211015.20.0/aarch64/fedora-coreos-34.20211015.20.0-qemu.aarch64.qcow2.xz

**Comment by rhatdan** [comment]
You are describing close source versus open source. And you may be right, but I think Podman and Fedora CoreOS will improve and still allow flexibility for users to us other distributions.  At one time Solaris worked better the Linux as well...

**Comment by afbjorklund** [comment]
I think some (including myself) was looking for a "slower" track, but it seems like that this would be covered by using Red Hat Enterprise Linux rather than Fedora ? The machine is handled by CoreOS and not by Podman, so I guess that's where to fix.

i.e. Podman doesn't link to a particular version of the server, but uses the latest tested and available from the _other_ project. Which means that any bug fix is first landed upstream, packaged in Fedora (RPM) and **then** included next fortnight (CoreOS)

Any user wanting to get bugs fixed faster than this monthly cadence, would probably have to run their own VM instead ?

Possibly there could be access to pre-releases or nightly builds, but eventually someone needs to support it as a "product"

----

There is talk about providing a Podman Desktop, so better to address those concerns there instead of in this bug report:

* Issue #11494

One way to get fixes faster is to run Fedora Cloud in the VM instead of Fedora CoreOS, as shown in [various guides](https://www.redhat.com/sysadmin/podman-clients-macos-windows).

Then you can administer and update the podman server yourself, through the package manager or even by building it ?
